tests/validation/tests/v3_api/test_cert_rotation.py

```diff
@@ -0,0 +1,124 @@
+import pytest
+import datetime
+from .common import *  # NOQA
+from rancher import ApiError
+
+#Gets the certificate expiration date and cert name. Stores them in a dict.
+def get_certs():
+    certs={}
+    client, cluster = get_user_client_and_cluster()
+    for key in cluster.certificatesExpiration:
+        certs[key]=parse_datetime(cluster.certificatesExpiration[key]["expirationDate"])
+    #Delete etcd node certs
+    remove=[]
+    for key in certs:
+        if "kube-etcd" in key:
+            remove.append(key)
+    for i in remove:
+        certs.pop(i)
+    #Get etcd node certs from node IP
+    nodes=get_etcd_nodes(cluster)
+    for node in nodes:
+        if node["labels"]["node-role.kubernetes.io/etcd"] == "true":
+            ipKey="kube-etcd-"+node["ipAddress"].replace(".","-")
+            certs[ipKey]=parse_datetime(cluster.certificatesExpiration[ipKey]["expirationDate"])
+    return certs
+
+#Turn expiration string into datetime
+def parse_datetime(expiration_string):
+    return datetime.datetime(int(expiration_string[:4]),int(expiration_string[5:7]),int(expiration_string[8:10]),int(expiration_string[11:13]),int(expiration_string[14:16]),int(expiration_string[17:19]))
+
+def compare_changed(certs2, time_now, changed):
+    if "kube-etcd" in changed:
+        for key in certs2:
+            if "kube-etcd" in key:
+                changed.append(key)
+        changed.remove("kube-etcd")
+    for i in changed:
+        #Checks that new certs expire 10 years after they were rotated
+        assert (certs2[i] >= (time_now + datetime.timedelta(days=3650)))
+
+def compare_unchanged(certs1, certs2, unchanged):
+    if "kube-etcd" in unchanged:
+        for key in certs2:
+            if "kube-etcd" in key:
+                unchanged.append(key)
+        unchanged.remove("kube-etcd")
+    for i in unchanged:
+        assert(certs1[i]==certs2[i])
+
+def rotate_certs():
+    client, cluster = get_user_client_and_cluster()
+    client.action(cluster, "rotateCertificates")
```

Comment:
This can be done as an action on the cluster  similar to `cluster.etcdBackups`. this can be found in test_bkp_restore.py file